src/engine/SCons/Tool/msvs.py

```diff
@@ -462,15 +477,25 @@ def __init__(self, dspfile, source, env):
 
         self.sconscript = env['MSVSSCONSCRIPT']
 
-        if 'cmdargs' not in env or env['cmdargs'] is None:
-            cmdargs = [''] * len(variants)
-        elif SCons.Util.is_String(env['cmdargs']):
-            cmdargs = [env['cmdargs']] * len(variants)
-        elif SCons.Util.is_List(env['cmdargs']):
-            if len(env['cmdargs']) != len(variants):
-                raise SCons.Errors.InternalError("Sizes of 'cmdargs' and 'variant' lists must be the same.")
+        def GetKeyFromEnv(env, key, variants):
```

Comment:
Move this outside of __init__()?

Reply:
This function was already using a locally-scoped function `AddConfig` so I wanted to follow that example. What are the best practices here?

Reply:
I guess that's fine then. I try to avoid locally scoped unless there's a reason, like we're using local variables inside the function lambda style..